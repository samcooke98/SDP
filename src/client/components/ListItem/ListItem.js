//You could call this file {ComponentName}.js as well, but you would have to ensure your imports are correct

import React from "react";
import style from "./web.css"

import Arrow from "react-icons/lib/md/keyboard-arrow-right.js"

export default class JournalBar extends React.Component {
    render() {
        return (
            <div style={{
                display: "flex",
                height: "60px",
                justifyContent: 'space-between',
                alignItems: 'center',
                marginBottom: "8px",
                cursor: "pointer",
                backgroundColor: this.props.active ? "#f1f1f1" : ""
            }}
                onClick={this.props.onClick}
            >
                <div style={{ width: "240px", height: "60px" }}>
                    <p className={style.title}> {this.props.title} </p>
                    <p className={style.caption}> {this.props.caption} </p>
                </div>
                <span style={{ width: "48px" }}>
                    <Arrow height={48} width={48} />
                </span>
            </div>
        )

    }
}


//In addition to CSS modules, you may wish to use: https://github.com/gajus/react-css-modules