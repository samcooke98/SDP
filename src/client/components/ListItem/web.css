.stack {}

p.title {
    font-family: "Roboto";
    font-weight: bold;
    font-size: 32px;
    margin: 0;
    color: #333333;
}

p.caption {
    font-family: "Roboto";
    font-size: 16px;
    margin: 0;
    color: #333333;
}