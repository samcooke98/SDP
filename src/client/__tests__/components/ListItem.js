import React from "react";
import renderer from "react-test-renderer";
// import ReactDOM from 'react-dom';

import ListItem from "../../components/ListItem/ListItem";


describe('React Component: ListItem', () => {
    it("matches a snapshot", () => {
        const tree = renderer.create(
            <ListItem />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})