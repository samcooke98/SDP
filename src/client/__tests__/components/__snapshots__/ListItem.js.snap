// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Component: ListItem matches a snapshot 1`] = `
<div
  onClick={undefined}
  style={
    Object {
      "alignItems": "center",
      "cursor": "pointer",
      "display": "flex",
      "height": "60px",
      "justifyContent": "space-between",
      "marginBottom": "8px",
    }
  }
>
  <div>
    <p
      className={undefined}
    >
       
       
    </p>
    <p
      className={undefined}
    >
       
       
    </p>
  </div>
  <svg
    fill="currentColor"
    height={48}
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width={48}
  >
    <g>
      <path
        d="m14.3 27.3l7.7-7.7-7.7-7.7 2.3-2.3 10 10-10 10z"
      />
    </g>
  </svg>
</div>
`;
